documentation/decisions/adr/2023-12-15-deliverable-reporting-strategy.md

```diff
@@ -0,0 +1,235 @@
+
+In light of these needs, the goal of this ADR is to:
+
+1. Recommend a consistent strategy for assigning issues to a parent 30k deliverable across reporting channels
+2. Recommend an approach to limiting the number of 30k deliverables that appear in our reports
+
+## Decision Drivers <!-- RECOMMENDED -->
+
+### Assigning issues to deliverables
+
+Our recommended strategy for assigning issues to a given 30k deliverable should consider the following criteria:
+
+- Users assigning an issue to a 30k deliverable should only need to update that value in one place
+- Users should be able to search or filter for the issues assigned to a given 30k deliverable
+- Users should be able to group by 30k deliverables in the sprint board and in [GitHub insights reporting][github-insights]
+- The relationship between issues and 30k deliverables should be consistently represented across all reports
+
+### Limiting the 30k deliverables included in reporting
+
+Our recommended approach to limiting the deliverables that are included in reporting should consider the following criteria:
+
+- The logic for including a 30k deliverable in reporting should be clear and easy to understand
+- Users should be able to add or remove a given 30k deliverable from reports without having to change the underlying logic
+- If necessary, users should be able to change the reporting logic without rewriting significant sections of the source code
+
+## Options considered
+
+### Assigning issues to deliverables
+
+- **Milestones:** Add the issue to a milestone that represents a 30k deliverable
+- **Labels:** Tag the issue with a label that represents a 30k deliverable
+- **Deliverable column:** Add a custom "deliverable" column to the GitHub projects with values representing each 30k deliverable
+- **Reserved phrase:** Use a reserved phrase to tag a 30k deliverable in the body of the issue
+
+### Limiting the 30k deliverables included in reporting
+
+- **Label:** Use a label to indicate when a deliverable should be included in reporting
+- **Status:** Use the status of the deliverable in the provisional roadmap project
+- **GitHub action:** Explicitly list the deliverables to include in the GitHub action for reporting
+
+## Decision outcomes <!-- REQUIRED -->
+
+### Assigning issues to deliverables
+
+We've decided to use a "deliverable" column to assign issues to a parent 30k deliverable. This involves:
+
+- creating a single select column in both the product roadmap and the sprint board
+- populating that column with options that represent each 30k deliverables in our roadmap
+- using that column to indicate the 30k deliverable that an issue is assigned to
+
+- **Positive outcomes**
+  - We can filter, group, and sort issues by deliverable in GitHub projects and GitHub's automated insight reporting.
+  - The engineering team can continue to use GitHub milestones to organize issues into units of work that are smaller than a 30k deliverable.
```

Comment:
On further reflection, I'm not sure this is entirely a positive. 

What *is* positive: being able to break down 30k into smaller chunks. 

What I'm not sure is positive: having the engineering team organize work in a way that is not necessarily nested within our overall 30k organizing framework. 

Let's say, in this option, the eng team organizes work according to milestones that are *not* cleanly mapped to 30k deliverables. 

For instance:
* Milestone A is 100% within 30k deliverable 1
* Milestone B is 50% within 30k deliverable 1 and 50% within deliverable 2 
* Milestone C is 66% within deliverable 2 and 34% within no deliverable 

Now any engineer or product person creating or updating a single ticket needs to track that ticket's placement within two separate and unrelated organizing hierarchies: both the milestone and the deliverable. 

It might make more sense to say: here's the 30k deliverable, and underneath that, you can create whatever milestones  you want that help organize that work, but all of them fit underneath the same 30k deliverable. (This would be "Milestone Organization B" as I refer to it in another comment on this PR.) 

Then when you create a new ticket, you choose a milestone for it, and by choosing the milestone it has also chosen the 30k deliverable for it. Now a engineer/product person only has to maintain one hierarchy for the ticket in their head, rather than figuring out two hierarchies for a single ticket.

Reply:
As discussed verbally with @widal001 , can we enforce a standard that every milestone is part of either 0 or 1 30Ks, but not more than one 30K? 

That will

Reply:
Added some of these notes in this commit: [fix: Incorporates PR comments](https://github.com/HHS/simpler-grants-gov/pull/926/commits/5831bfabf360fca818dc281ddd31129b6152a7bd)